fiftyone/core/collections.py

```diff
@@ -4980,7 +4995,7 @@ def _record_list_field(field_name):
     if root_field_name in ("id", "_id"):
         root_field = None
     elif root_field_name not in schema:
-        if not allow_missing:
+        if not allow_missing and not root_field_name.startswith("_"):
```

Comment:
per our online discussion: I think the best approach here is to add an `_allow_missing` parameter to `SetField()` so and use that instead of the `root_field` check here